src/settings/Settings.cpp

```diff
@@ -0,0 +1,585 @@
+//Copyright (c) 2018 Ultimaker B.V.
+//CuraEngine is released under the terms of the AGPLv3 or higher.
+
+#include <cctype>
+#include <fstream>
+#include <stdio.h>
+#include <sstream> // ostringstream
+#include <regex> // regex parsing for temp flow graph
+#include <string> //Parsing strings (stod, stoul).
+
+#include "Settings.h"
+#include "types/AngleDegrees.h" //For angle settings.
+#include "types/AngleRadians.h" //For angle settings.
+#include "types/Duration.h" //For duration and time settings.
+#include "types/LayerIndex.h" //For layer index settings.
+#include "types/Ratio.h" //For ratio settings and percentages.
+#include "types/Temperature.h" //For temperature settings.
+#include "types/Velocity.h" //For velocity settings.
+#include "../Application.h" //To get the extruders.
+#include "../utils/logoutput.h"
+
+namespace cura
+{
+
+void Settings::add(const std::string& key, const std::string value, ExtruderTrain* limit_to_extruder)
+{
+    settings.insert(std::pair<std::string, Setting>(key, Setting(value, limit_to_extruder)));
```

Comment:
instead of `settings.insert(std::pair<...>(...)` you could just do `settings.emplace(...)`

Reply:
Done. That's indeed much nicer.